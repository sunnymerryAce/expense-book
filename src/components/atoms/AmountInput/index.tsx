import React from "react";
import TextField from "@material-ui/core/TextField";
import NumberFormat from "react-number-format";
import useStyles from "./style";

export interface AmountInputProps {
  label: string;
  error: boolean;
  defaultValue?: string;
  helperText?: string;
  handleChange?: (value: string) => void;
  className?: string;
  disabled?: boolean;
}

const AmountInput: React.FC<AmountInputProps> = (props) => {
  const classes = useStyles();

  const handleChange = ({ value }: { value: string }): void => {
    props.handleChange && props.handleChange(value);
  };

  return (
    <NumberFormat
      prefix="Â¥"
      type="tel"
      thousandSeparator={true}
      isNumericString={true}
      customInput={TextField}
      variant="outlined"
      onValueChange={handleChange}
      defaultValue={props.defaultValue || ""}
      error={props.error}
      helperText={props.helperText}
      label={props.label}
      disabled={props.disabled || false}
      className={`AmountInput ${props.className && props.className} ${
        classes.root
      }`}
    />
  );
};

export default AmountInput;
