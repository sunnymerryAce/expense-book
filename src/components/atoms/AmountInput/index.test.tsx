import React from "react";
import AmountInput from "./";
import { shallow, mount } from "enzyme";

describe("<AmountInput />", () => {
  let container: any = null;
  const props = {
    label: "Budget",
    defaultValue: "25000",
    error: false,
    handleChange: jest.fn(),
  };

  afterEach(() => {
    container.unmount();
    container = null;
  });

  test("should match the snapshot", () => {
    container = shallow(<AmountInput {...props} />);
    expect(container.html()).toMatchSnapshot();
  });

  test("should have proper label", () => {
    container = mount(<AmountInput {...props} />);
    expect(container.find("label.MuiInputLabel-root").text()).toBe(props.label);
  });

  test("should have proper default value", () => {
    container = mount(<AmountInput {...props} />);
    expect(container.find("input.MuiInputBase-input").get(0).props.value);
  });

  test("should be disabled", () => {
    const props = {
      label: "Budget",
      defaultValue: "25000",
      error: false,
      handleChange: jest.fn(),
      disabled: true,
    };
    container = mount(<AmountInput {...props} />);
    expect(
      container.find("input.MuiInputBase-input").get(0).props.disabled
    ).toBe(true);
  });

  test("should be error", () => {
    const props = {
      label: "Budget",
      defaultValue: "25000",
      error: true,
      helperText: "helperText",
      handleChange: jest.fn(),
    };
    container = mount(<AmountInput {...props} />);
    expect(container.exists(".MuiInputBase-root.Mui-error")).toBe(true);
  });

  test("should have proper error text", () => {
    const props = {
      label: "Budget",
      defaultValue: "25000",
      error: true,
      helperText: "helperText",
      handleChange: jest.fn(),
    };
    container = mount(<AmountInput {...props} />);
    expect(container.find(".MuiFormHelperText-root").text()).toBe(
      props.helperText
    );
  });
});
