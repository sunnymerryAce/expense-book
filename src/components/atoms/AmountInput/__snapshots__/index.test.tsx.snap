// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AmountInput /> should match the snapshot 1`] = `"<div class=\\"MuiFormControl-root MuiTextField-root AmountInput undefined makeStyles-root-1\\" inputMode=\\"numeric\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled\\" data-shrink=\\"true\\">Budget</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl\\"><input type=\\"tel\\" aria-invalid=\\"false\\" value=\\"Â¥25,000\\" maxLength=\\"12\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-4 PrivateNotchedOutline-legendNotched-5\\"><span>Budget</span></legend></fieldset></div></div>"`;
